incubator/group/types.proto

```diff
@@ -0,0 +1,181 @@
+    bytes admin = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    uint64 group = 2 [(gogoproto.casttype) = "GroupID"];
+    repeated Member member_updates = 3; // set power to 0 to remove
+}
+
+message MsgUpdateGroupAdmin {
+    bytes admin = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    uint64 group = 2 [(gogoproto.casttype) = "GroupID"];
+    bytes new_admin = 3 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+}
+
+message MsgUpdateGroupComment {
+    bytes admin = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    uint64 group = 2 [(gogoproto.casttype) = "GroupID"];
+    string comment = 3;
+}
+
+message Member {
+    bytes address = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    string power = 2 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec"];
+    string comment = 3;
+}
+
+//
+// Group Accounts
+//
+
+message MsgCreateGroupAccount {
+    bytes admin = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    uint64 group = 2 [(gogoproto.casttype) = "GroupID"];
+    google.protobuf.Any decision_policy = 3;
+    string comment = 4;
+}
+
+message MsgUpdateGroupAccountAdmin {
+    bytes admin = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    bytes group_acount = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    bytes new_admin = 3 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+}
+
+message MsgUpdateGroupAccountDecisionPolicy {
+    bytes admin = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    bytes group_acount = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    google.protobuf.Any decision_policy = 3;
+}
+
+message MsgUpdateGroupAccountComment {
+    bytes admin = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    bytes group_acount = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    string comment = 3;
+}
+
+message ThresholdDecisionPolicy {
+    string threshold = 1 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec"];
+    google.protobuf.Duration min_voting_window = 2;
+    google.protobuf.Duration max_voting_window = 3;
+}
+
+//
+// Proposals and Voting
+//
+
+message MsgPropose {
+    bytes group_account = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    // Proposers signatures will be counted as yes votes
+    repeated bytes proposers = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    repeated google.protobuf.Any msgs= 3; // []sdk.Msg
+    string comment = 4;
+    // ExecNow will attempt to execute the proposal immediately based on
+    // the proposers signing the transaction
+    bool exec_now = 5;
+}
+
+enum Choice {
+    NO = 0;
```

Comment:
PP: Let's reserve `0` for INVALID that we fail if not set and add prefixes to the names.

I would suggest we find adopt a styleguide for protobuf (for example [this](https://github.com/uber/prototool/blob/dev/style/README.md#enums) ) and run https://github.com/uber/prototool linter during CI. We had https://github.com/iov-one/prototool-docker that standardized the code generation as well.

Reply:
Cosmos SDK is using buf (a recommended replacement for prototool) and we should use as well. Added `UNKNOWN = 0`